File: src/055/p055.py
from src.utils.py.math import is_palindrome


def is_lychrel_number(num: int) -> bool:
    for _ in range(50):
        num += int(str(num)[::-1])
        if is_palindrome(num):
            return False
    return True


def solve() -> int:
    count = 0
    for num in range(1, 10000):
        if is_lychrel_number(num):
            count += 1
    return count

File: src/055/p055.go
package problem055

import (
	"math/big"
	"strconv"

	"github.com/sunilbpandey/project-euler/src/utils/go/strutils"
)

func reverse(n *big.Int) *big.Int {
	r, _ := big.NewInt(0).SetString(strutils.Reverse(n.String()), 10)
	return r
}

func isLychrel(num int) bool {
	n := big.NewInt(int64(num))
	for i := 0; i < 50; i++ {
		n.Add(n, reverse(n))
		if strutils.IsPalindrome(n.String()) {
			return false
		}
	}
	return true
}

func Solve() string {
	count := 0
	for n := 1; n < 10000; n++ {
		if isLychrel(n) {
			count++
		}
	}
	return strconv.Itoa(count)
}
